idb/postgres/postgres.go

```diff
@@ -2495,3 +2495,58 @@ func (db *IndexerDb) SetNetworkState(genesis bookkeeping.Genesis) error {
 	}
 	return db.setNetworkState(nil, &networkState)
 }
+
+// DeleteTransactions removes old transactions
+// keep is the number of rounds to keep in db
+func (db *IndexerDb) DeleteTransactions(ctx context.Context, keep uint64, timeout time.Duration) (int64, error) {
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	// delete old transactions and update metastate
+	deleteTxns := func() (int64, error) {
+		// start a transaction
+		tx, err2 := db.db.BeginTx(ctx, pgx.TxOptions{})
+		if err2 != nil {
+			return 0, fmt.Errorf("deleteTxns(): %w", err2)
+		}
+		defer tx.Rollback(ctx)
+
+		db.log.Infof("deleteTxns(): keeping round %d and later", keep)
+		// delete query
+		query := "DELETE FROM txn WHERE round < $1"
+		cmd, err2 := tx.Exec(ctx, query, keep)
+		if err2 != nil {
+			return 0, fmt.Errorf("deleteTxns(): transaction delete err %w", err2)
+		}
+		t := time.Now().UTC()
+		// update metastate
+		status := types.DeleteStatus{
+			// format time, "2006-01-02T15:04:05Z07:00"
+			LastPruned:  t.Format(time.RFC3339),
+			OldestRound: keep,
+		}
+		if err2 != nil {
+			return 0, fmt.Errorf("deleteTxns(): transaction delete err %w", err2)
+		}
+		err2 = db.setMetastate(tx, schema.DeleteStatusKey, string(encoding.EncodeDeleteStatus(&status)))
```

Comment:
:+1:  Nice touch adding status to the metastate. We sometimes get into trouble reverse engineering a weird DB state, we should do stuff like this more often.